pyro/distributions/transforms/neural_autoregressive.py

```diff
@@ -82,33 +82,6 @@ def leaky_relu():
     return LeakyReLUTransform()
 
 
-class TanhTransform(Transform):
-    r"""
-    Bijective transform via the mapping :math:`y = \text{tanh}(x)`.
-    """
-    domain = constraints.real
-    codomain = constraints.interval(-1., 1.)
-    bijective = True
-    sign = +1
-
-    @staticmethod
-    def atanh(x):
-        return 0.5 * (x.log1p() - (-x).log1p())
-
-    def __eq__(self, other):
-        return isinstance(other, TanhTransform)
-
-    def _call(self, x):
-        return torch.tanh(x)
-
-    def _inverse(self, y):
-        eps = torch.finfo(y.dtype).eps
-        return self.atanh(y.clamp(min=-1. + eps, max=1. - eps))
-
-    def log_abs_det_jacobian(self, x, y):
-        return - 2. * (x - math.log(2.) + F.softplus(- 2. * x))
```

Comment:
@stefanwebb note that the torch version does not clamp and instead suggests using `cache_size=1`

https://github.com/pytorch/pytorch/blob/99349393ba9bc9b21a6fcb5c7cefd1738f9a6b98/torch/distributions/transforms.py#L431-L438

Reply:
I think it should be fine to remove the `Tanh` from Pyro and use the one from PyTorch that doesn't do clamping